### PR TITLE
GH1065 timestamp freq argument and DataFrame/Series.mask

### LINKED ISSUE #1065: Version 2.0 Migration Timestamp/Mask
- [x] Removed deprecated Timestamp.freq, Timestamp.freqstr and argument freq from the [Timestamp](https://pandas.pydata.org/docs/reference/api/pandas.Timestamp.html#pandas.Timestamp) constructor and [Timestamp.fromordinal()](https://pandas.pydata.org/docs/reference/api/pandas.Timestamp.fromordinal.html#pandas.Timestamp.fromordinal)
- [ ] The other argument in [DataFrame.mask()](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.mask.html#pandas.DataFrame.mask) and [Series.mask()](https://pandas.pydata.org/docs/reference/api/pandas.Series.mask.html#pandas.Series.mask) now defaults to no_default instead of np.nan consistent with [DataFrame.where()](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.where.html#pandas.DataFrame.where) and [Series.where()](https://pandas.pydata.org/docs/reference/api/pandas.Series.where.html#pandas.Series.where).

### PR DESCRIPTION
**Pull request by loicdiridollou**
- [x] Closes #1065 
- [x] Tests added: Please use `assert_type()` to assert the type of any return value

### CODE REVIEW

File: pandas-stubs/_libs/tslibs/timestamps.pyi [hunk 1]
```diff
@@ -59,9 +59,6 @@ class Timestamp(datetime):
     def __new__(
         cls,
         ts_input: np.integer | float | str | _date | datetime | np.datetime64 = ...,
-        # Freq is deprecated but is left in to allow code like Timestamp(2000,1,1)
-        # Removing it would make the other arguments position only
-        freq: int | str | BaseOffset | None = ...,
         tz: str | _tzinfo | int | None = ...,
         unit: str | int | None = ...,
```
Comment (Dr-Irv) [inline pandas-stubs/_libs/tslibs/timestamps.pyi:63]: We need to clean this up based on the current API.  `tz` and `unit` should move after `nanosecond`

File: pandas-stubs/_libs/tslibs/timestamps.pyi [hunk 2]
```diff
@@ -71,9 +68,9 @@ class Timestamp(datetime):
         minute: int | None = ...,
         second: int | None = ...,
         microsecond: int | None = ...,
+        *,
         nanosecond: int | None = ...,
         tzinfo: _tzinfo | None = ...,
```
Comment (Dr-Irv) [inline pandas-stubs/_libs/tslibs/timestamps.pyi:73]: `tzinfo` should move before the asterisk.

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

File: pandas-stubs/_libs/tslibs/timestamps.pyi (hunk 2 above)
Comment (loicdiridollou) [reply pandas-stubs/_libs/tslibs/timestamps.pyi]: I have checked with the pandas version and realigned all the arguments properly.

**Review by Dr-Irv — APPROVED** [review]
thanks @loicdiridollou

**Pull request merged**